### ISSUE TITLE
No such process found!

### ISSUE THREAD

**Issue by rahilwazir** [neekey/ps#7]
Following the code:

    let childWorker = cp.fork(`${__dirname}/../somefile`);

    console.log(childWorker.pid);
    // 12345

    ps.lookup({ pid: childWorker.pid }, (err, resultList) => {
        console.log(resultList);
        // []
    });

Why it didn't work? I also randomly picked pid from `ps -A` and it still didn't work.

**Comment by hitchcott** [comment]
Weird, I am also having this issue. It just doesn't work!

```coffeescript
ps.lookup {command: 'node'}, (err,res) ->
  console.log 'ps.lookup', err, res # always []
```

**Comment by jamietre** [comment]
This doesn't work at all for me in Winows either.

**Comment by sleman** [comment]
I believe issue is with the format method.

On Mac, I run `ps` and I get:
`/Users/bin/ffmpeg-f:v mpegts -fflags nobuffer -probesize 18192 -i udp://8.8.8.8:9000 -ignore_unknown -vf fps=20 /tmp/`

However, when running ps.lookup, that process isn't found.

**Comment by hitchcott** [comment]
@sleman you're correct - it seemed to be a problem with TableParse - it wasn't parsing the PID correctly

**Comment by neekey** [comment]
Hi, guys, I'm sure this is caused by https://github.com/neekey/table-parser/pull/3 , I will fix it soon.

**Comment by ricovitch** [comment]
Does not work for me on windows.
I see @hitchcott made a patch, but i prefer to seek for another solution.

**Comment by codeandcats** [comment]
Yep TableParse is the issue. It is expecting line endings to be \n when they can be \r (even on windows).

**Comment by leegee** [comment]
:+1:

**Comment by jamietre** [comment]
Or just use sindre sorhus' package which works: https://github.com/sindresorhus/ps-list 

I'm all for alternatives, but this code has been broken since I first tried it 6 months ago. You should make a note in the readme that it doesn't work so people don't keep wasting time, until if/when it is fixed.

**Comment by codeandcats** [comment]
Awesome, thanks for the suggestion @jamietre!

**Comment by leegee** [comment]
On the `table-parser` issue page, nfantone commented on 2 Jan this year, `ps-node` doesn't even need a table parser. Thanks, @jamietre, for your suggestion - I'll try that.

**Comment by yetzt** [comment]
i noticed, i don't get a result when calling with an interger as pid paremeter, but converting it to a string it works just well.

**Comment by yetzt** [comment]
ping @neekey.

**Comment by yetzt** [comment]
well, i sent @neekey an email, but so far no response. i implemented the functionality i needed myself [in this module](https://www.npmjs.com/package/piss), feel free to use it.

**Comment by PaulTerryGameDev** [comment]
Does this work for Windows 10 64-bit?

**Comment by PaulTerryGameDev** [comment]
(It doesn't seem to work for me in Windows 10 64-bit..is not finding any processes by ID (using the first example in the help)

**Comment by yetzt** [comment]
@PaulTerryGameDev this uses the `ps` command found on unix operating systems. you could write something that uses `tasklist` and parse it's output.

**Comment by neekey** [comment]
Sorry guys! I'm back! I just rewrite [table-parser](https://github.com/neekey/table-parser) with [connected-domain](https://github.com/neekey/connected-domain), and I create a new branch of PS using the latest version of table-parser.

The parser problem should be fixed, but I need you to help me test with it, install it through github:

```
npm install https://github.com/neekey/ps.git#0.1.0
```

@yetzt your PR is included

@PaulTerryGameDev I haven't got a chance to try win 10, can you open a new issue for it?

**Comment by neekey** [comment]
Hi guys, I have published version 0.1.0, and I will focus on these two issues( https://github.com/neekey/table-parser/issues/4, https://github.com/neekey/ps/issues/10 ) ,feel free to reopen this issue if you encounter any problem related.

**Comment by Luiz-N** [comment]
I'm having this issue now with latest version

**Comment by neekey** [comment]
Hi @Luiz-N, could you please provide more details like:
- your OS version
- NodeJS version
- the code you were running

thank you

2016年5月7日星期六，Luiz-N <notifications@github.com> 写道：

> I'm having this issue now with latest version
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly or view it on GitHub
> <https://github.com/neekey/ps/issues/7#issuecomment-217535259>
>


-- 
发自移动版 Gmail

**Comment by cloudtaxi** [comment]
I have the same issue with latest version
    └── ps-node@0.1.0 

    [root@bigbn site]# node ets.js 
    No such process found!
    [root@bigbn site]# node --version
    v6.1.0
    [root@bigbn site]# uname -a
    Linux bigbn 4.5.2-1-ARCH #1 SMP PREEMPT Thu Apr 21 18:21:27 CEST 2016 x86_64 GNU/Linux

I'm using code from first example with my existsting pid

**Comment by neekey** [comment]
@cloudtaxi ps-node has already passed test on Mac and Linux through https://travis-ci.org/neekey/ps, and I tested it with NodeJS 6.1.0 on my Mac, it works. Can you install the latest version of 0.1.1, run its test cases with `npm test` and tell me the result ?

Thank you

**Comment by SummerWish** [comment]
I have the same issue as @cloudtaxi 
1. in OS X: `ps l` will not return a full list of processes.
2. `ps aux` works, however the columns are incorrectly-mapped, may due to Chinese characters

**Comment by SummerWish** [comment]
It may be much more easy to parse result if you use formatted output as https://github.com/yibn2008/find-process/blob/master/lib/find_process.js#L27

**Comment by neekey** [comment]
@SummerWish if your problem is caused by multiple-byte characters, then it is a know issue with table-parser https://github.com/neekey/table-parser/issues/4

However, the link you provided does not help much as you can see from the manual for `ps`:

```shell
-w      Use 132 columns to display information, instead of the default which is your window size.  If the -w option is specified more than once, ps will use as many columns as necessary without regard for your
             window size.  When output is not to a terminal, an unlimited number of columns are always used.
-o      Display information associated with the space or comma separated list of keywords specified.  Multiple keywords may also be given in the form of more than one -o option.  Keywords may be appended with
             an equals (`=') sign and a string.  This causes the printed header to use the specified string instead of the standard header.  If all keywords have empty header texts, no header line is written.
```

`-w` and `-o` really do not help to solve this problem, and to be honest, I do not have a good idea to deal with it right now. But you can avoid this problem by not using option "u" ( notice not "-u", "u" means to show detailed information, which will include a `STARTED` field, which may be output as local string ) or "-f" which quite does the same thing.

Another idea I just came out is maybe there's a way I can enforce the PS command to output a specified language like English, but I am not sure if it can be done. 

How this can give you a little help : )

**Comment by SummerWish** [comment]
Hi, thanks for your explanation! How about just split each row by whitespace? Since you can use `-o` to put the arg column as the last column and project only part of the columns, other columns should not contain any whitespace.

**Comment by neekey** [comment]
@SummerWish the situation is much more complicated than you may have thought, for example, for some rows their certain field may be blank, and another case is that for COMMAND filed, they will include space to separate their command and arguments like:

```
System/Library/Frameworks/CoreServices.framework/Frameworks/Metadata.framework/Versions/A/Support/mdworker -s mdworker -c MDSImporterWorker -m com.apple.mdworker.single
```

and in windows you may not be able to control the position for COMMAND field, and not even in some case you will encounter strings like below:

```
"hello world"
```

Hope this makes sense to you

**Comment by SummerWish** [comment]
Besides, `ux` breaks this utility in OS X:

```js
ps.lookup({ psargs: 'ux' }, ...);
```

outputs:

```json
{ pid: 'Breezewish',
    command: 'node',
    arguments: [ 't.js' ],
    ppid: undefined }
```

which is not correct

**Comment by SummerWish** [comment]
Wow. Windows is a problem that I didn't considered :(

**Comment by MikiDavid** [comment]
Hi @neekey,

I had this issue now:

Attempt to found an active process by its name: NearoHostServer.exe - Fail each time and got `[]`.
When at command I write nearohostserver.exe - Works like a charm and giving back the details.

My code:
**Working:**

```
var ps = require('ps-node');

ps.lookup({
	command: 'nearohostserver.exe',
	arguments: ''	
}, function(error, resultList){
	console.log(resultList);
        // [ { pid: '1104',
        // command: 'c:\\program files (x86)\\mce-100\\nearohostserver.exe',
        // arguments: [ '5F975E550A3FD20DDCA48B38EAF586F2', '/worker' ],
        // ppid: undefined } ]
})
```

**Not Working:**
```
var ps = require('ps-node');

ps.lookup({
	command: 'NearoHostServer.exe',
	arguments: ''	
}, function(error, resultList){
	console.log(resultList);
        // []
})
```

**Comment by neekey** [comment]
@MikiDavid thanks for submiting this bug, it seems like a problem caused by string matching, should be fixed easily.

**Comment by neekey** [comment]
@MikiDavid I just published 0.1.2, it should fix this bug.

**Comment by xmsz-stu** [comment]
not working
